SofaKernel/framework/sofa/core/topology/BaseMeshTopology.h

```diff
@@ -132,7 +132,9 @@ class SOFA_CORE_API BaseMeshTopology : public core::topology::Topology
     virtual const Triangle getTriangle(TriangleID i) { return getTriangles()[i]; }
     virtual const Quad getQuad(QuadID i)             { return getQuads()[i]; }
     virtual const Tetra getTetrahedron(TetraID i)    { return getTetrahedra()[i]; }
-    virtual const Hexa getHexahedron(HexaID i)       { return getHexahedra()[i]; }
+    virtual const Hexa getHexahedron(HexaID i)       { return getHexahedra()[i]; }   
+	   
+    virtual sofa::core::topology::TopologyObjectType getTopologyType() const = 0 ;
```

Comment:
This method is not actually used, and I haven't checked why it was added exactly, but it looks like the main benefit is to get rid of codes that include and use chains of dynamic casts to *SetTopologyContainer to discover which type of topology it is, and to get expose the "UpperTopology" info already computed in MeshTopology that have a similar (undocumented) semantic. In case of *SetTopologyContainer, the semantic is clear, as you cannot mix triangles and quads. It is indeed unclear for MeshTopology, but that is nothing new (as this change is not computing it, just exposing it in a unified way).
I would rather have this method without comment and have a todo somewhere to clarify MeshTopology::UpperTopology, document it and this method, and add unit test(s), than to add an unclear comment here...